blob/blob.go

```diff
@@ -588,19 +601,43 @@ func Open(ctx context.Context, urlstr string) (*Bucket, error) {
 	return NewBucket(drv), nil
 }
 
+// wrappedError is used to wrap all errors returned by drivers so that users
+// are not given access to provider-specific errors.
+type wrappedError struct {
```

Comment:
This seems quite reasonable. I don't think your EOF check is fragile; that is the only pseudo-error in this part of the standard library, so it makes sense to special-case it.

How do you feel about adding an `Unwrap` method? Or maybe an `As` method? That will provide an escape hatch for provider-aware error testing. See https://go.googlesource.com/proposal/+/master/design/go2draft-error-inspection.md.

Reply:
Good idea! I filed #750.